### ISSUE TITLE
Issues updating to 8.10 on Rockport64 board that was running fine

### ISSUE THREAD

**Issue by Wilford888** [MichaIng/DietPi#5845]
#### Details:
- Date           | Sat Oct 29 19:56:37 BST 2022
- DietPi version | v8.9.2 (MichaIng/master)
- Image creator  | DietPi Core Team
- Pre-image      | from scratch
- Hardware       | Quartz64 (aarch64) (ID=49)
- Kernel version | `Linux DietPiTR 5.19.0-rc1 #1 SMP PREEMPT Fri Jul 15 09:39:18 UTC 2022 aarch64 GNU/Linux`
- Distro         | bullseye (ID=6)
- Command        | `getent hosts dns9.quad9.net`
- Exit code      | 2
- Software title | DietPi-Update
#### Steps to reproduce:

1. I've tried twice to update to 8.10.  Once from a running unit, and when that failed, I reflashed the eMMC with the latest image and rebooted.  On reboot, the unit tried to update automatically and again hung.  The update seems to finish but I cannot log in to the unit after the update.  This has happened twice in a row.
2. ...
#### Expected behaviour:

- Well, normally the updates work fine.
#### Actual behaviour:

- Unit seems to update, restarts, and then I can no longer access the board.
#### Extra details:

- I tried a complete reflash of the eMMC, after the auto-update ran, same result.  Updates seems to happen just fine.  Log out, cannot log back in.
#### Additional logs:

```
 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Completed

[ INFO ] DietPi-Update | Current version : v8.10.2
[ INFO ] DietPi-Update | Latest version  : v8.10.2
[  OK  ] DietPi-Survey | Setting in /boot/dietpi.txt adjusted: SURVEY_OPTED_IN=0
[  OK  ] DietPi-Survey | Purging survey data
[ INFO ] DietPi-Update | A reboot is done to finalise the kernel upgrade
Connection to 192.xxx.xxx.127 closed by remote host.
Connection to 192.xxx.xxx.127 closed.
[It is after this point that I can no longer login via terminal.]
```

<details>
  <summary>Logfile attached. Click to expand!</summary>

```
[Entire screen output after the reflash to eMMC and restart:]

 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Checking for available DietPi update

[  OK  ] DietPi-Update | Checking IPv4 network connectivity
[  OK  ] DietPi-Update | Checking DNS resolver
[  OK  ] DietPi-TimeSync | systemctl stop systemd-timesyncd
[  OK  ] DietPi-TimeSync | mkdir -p /run/systemd/timesync
[ INFO ] DietPi-Update | Getting latest version from: https://raw.githubusercontent.com/MichaIng/DietPi/master/.update/version
[  OK  ] DietPi-Update | Got valid latest version: 8.10.2
[  OK  ] DietPi-Update | Update available:
[ INFO ] DietPi-Update | Current version : v8.9.2
[ INFO ] DietPi-Update | Latest version  : v8.10.2

 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Checking for update pre-requirements

[  OK  ] DietPi-Update | DietPi-Userdata validation: /mnt/dietpi_userdata
[  OK  ] DietPi-Update | Free space check: path=/ | available=114169 MiB | required=100 MiB
[ SUB1 ] DietPi-Services > stop
[ INFO ] DietPi-Services | skip : cron (due to mask)

 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Applying pre-patches

[  OK  ] DietPi-Update | Downloading pre-patches
[  OK  ] DietPi-Update | Applying execute permission
[  OK  ] DietPi-Update | Successfully applied pre-patches

 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Upgrading APT packages

[ INFO ] DietPi-Update | APT update, please wait...
Get:2 https://deb.debian.org/debian bullseye InRelease [116 kB]
Get:3 https://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 https://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Get:5 https://deb.debian.org/debian bullseye-backports InRelease [49.0 kB]
Get:6 https://deb.debian.org/debian bullseye/non-free arm64 Packages [72.9 kB]
Get:7 https://deb.debian.org/debian bullseye/main arm64 Packages [8071 kB]
Get:8 https://deb.debian.org/debian bullseye/contrib arm64 Packages [41.0 kB]
Get:9 https://deb.debian.org/debian bullseye-updates/main arm64 Packages [12.0 kB]
Get:10 https://deb.debian.org/debian-security bullseye-security/main arm64 Packages [191 kB]
Get:11 https://deb.debian.org/debian bullseye-backports/main arm64 Packages [354 kB]
Get:12 https://deb.debian.org/debian bullseye-backports/contrib arm64 Packages [3840 B]
Get:13 https://deb.debian.org/debian bullseye-backports/non-free arm64 Packages [9868 B]
Get:1 http://mirrors.jevincanders.net/armbian/apt bullseye InRelease [22.4 kB]
Get:14 http://mirrors.jevincanders.net/armbian/apt bullseye/main arm64 Packages [533 kB]
Fetched 9568 kB in 10s (969 kB/s)
Reading package lists...
[  OK  ] DietPi-Update | APT update
[ INFO ] DietPi-Update | APT upgrade, please wait...
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages will be upgraded:
  armbian-firmware libc-bin libc-l10n libc6 libksba8
  linux-dtb-current-rockchip64 linux-image-current-rockchip64
  linux-u-boot-rockpro64-current locales tzdata
10 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 72.0 MB of archives.
After this operation, 110 kB of additional disk space will be used.
Get:5 https://deb.debian.org/debian bullseye-updates/main arm64 libc6 arm64 2.31-13+deb11u5 [2460 kB]
Get:6 https://deb.debian.org/debian bullseye-updates/main arm64 libc-bin arm64 2.31-13+deb11u5 [743 kB]
Get:7 https://deb.debian.org/debian bullseye-updates/main arm64 tzdata all 2021a-1+deb11u7 [284 kB]
Get:8 https://deb.debian.org/debian bullseye-updates/main arm64 libc-l10n all 2.31-13+deb11u5 [865 kB]
Get:9 https://deb.debian.org/debian bullseye-updates/main arm64 locales all 2.31-13+deb11u5 [4086 kB]
Get:10 https://deb.debian.org/debian-security bullseye-security/main arm64 libksba8 arm64 1.5.0-3+deb11u1 [115 kB]
Get:2 http://armbian.tnahosting.net/apt bullseye/main arm64 linux-dtb-current-rockchip64 arm64 22.08.8 [380 kB]
Get:1 http://armbian.chi.auroradev.org/apt bullseye/main arm64 armbian-firmware all 22.08.6 [9409 kB]
Get:3 http://armbian.tnahosting.net/apt bullseye/main arm64 linux-image-current-rockchip64 arm64 22.08.8 [53.2 MB]
Get:4 http://armbian.tnahosting.net/apt bullseye/main arm64 linux-u-boot-rockpro64-current arm64 22.08.6 [436 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 72.0 MB in 12s (5957 kB/s)
(Reading database ... 17684 files and directories currently installed.)
Preparing to unpack .../libc6_2.31-13+deb11u5_arm64.deb ...
Unpacking libc6:arm64 (2.31-13+deb11u5) over (2.31-13+deb11u4) ...
Setting up libc6:arm64 (2.31-13+deb11u5) ...
(Reading database ... 17684 files and directories currently installed.)
Preparing to unpack .../libc-bin_2.31-13+deb11u5_arm64.deb ...
Unpacking libc-bin (2.31-13+deb11u5) over (2.31-13+deb11u4) ...
Setting up libc-bin (2.31-13+deb11u5) ...
(Reading database ... 17684 files and directories currently installed.)
Preparing to unpack .../0-tzdata_2021a-1+deb11u7_all.deb ...
Unpacking tzdata (2021a-1+deb11u7) over (2021a-1+deb11u6) ...
Preparing to unpack .../1-libc-l10n_2.31-13+deb11u5_all.deb ...
Unpacking libc-l10n (2.31-13+deb11u5) over (2.31-13+deb11u4) ...
Preparing to unpack .../2-locales_2.31-13+deb11u5_all.deb ...
Unpacking locales (2.31-13+deb11u5) over (2.31-13+deb11u4) ...
Preparing to unpack .../3-armbian-firmware_22.08.6_all.deb ...
Unpacking armbian-firmware (22.08.6) over (22.08.4) ...
Preparing to unpack .../4-libksba8_1.5.0-3+deb11u1_arm64.deb ...
Unpacking libksba8:arm64 (1.5.0-3+deb11u1) over (1.5.0-3) ...
Preparing to unpack .../5-linux-dtb-current-rockchip64_22.08.8_arm64.deb ...
Unpacking linux-dtb-current-rockchip64 (22.08.8) over (22.08.4) ...
Preparing to unpack .../6-linux-image-current-rockchip64_22.08.8_arm64.deb ...
Unpacking linux-image-current-rockchip64 (22.08.8) over (22.08.4) ...
Removing obsolete initramfs images
Preparing to unpack .../7-linux-u-boot-rockpro64-current_22.08.6_arm64.deb ...
Unpacking linux-u-boot-rockpro64-current (22.08.6) over (22.08.4) ...
Setting up libksba8:arm64 (1.5.0-3+deb11u1) ...
Setting up libc-l10n (2.31-13+deb11u5) ...
Setting up linux-image-current-rockchip64 (22.08.8) ...
Removing obsolete initramfs images
removed '/boot/initrd.img-5.15.72-rockchip64'
removed '/boot/uInitrd-5.15.72-rockchip64'
update-initramfs: Generating /boot/initrd.img-5.15.75-rockchip64
update-initramfs: Converting to U-Boot format
Image Name:   uInitrd
Created:      Sat Oct 29 22:13:40 2022
Image Type:   AArch64 Linux RAMDisk Image (gzip compressed)
Data Size:    11332656 Bytes = 11067.05 KiB = 10.81 MiB
Load Address: 00000000
Entry Point:  00000000
Setting up locales (2.31-13+deb11u5) ...
Generating locales (this might take a while)...
  C.UTF-8... done
Generation complete.
Setting up tzdata (2021a-1+deb11u7) ...

Current default time zone: 'Europe/London'
Local time is now:      Sat Oct 29 22:13:46 BST 2022.
Universal Time is now:  Sat Oct 29 21:13:46 UTC 2022.
Run 'dpkg-reconfigure tzdata' if you wish to change it.

Setting up armbian-firmware (22.08.6) ...
Processing triggers for libc-bin (2.31-13+deb11u5) ...

─────────────────────────────────────────────────────

[  OK  ] DietPi-Update | Downloading update archive
[  OK  ] DietPi-Update | Unpacking update archive
[  OK  ] DietPi-Update | Removing unused files
[  OK  ] DietPi-Update | Hardening update archive mode
[  OK  ] DietPi-Update | Installing new DietPi scripts
[  OK  ] DietPi-Update | Installing new DietPi system files
[ SUB1 ] DietPi-Set_software > verify_dietpi.txt
[  OK  ] DietPi-Set_software | Downloading current dietpi.txt
[  OK  ] verify_dietpi.txt  | Completed

 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Applying incremental patches

[ INFO ] DietPi-Update | Current version : v8.9.2
[ INFO ] DietPi-Update | Latest version  : v8.10.2
[ INFO ] DietPi-Patch | Patching to DietPi v8.10...
[  OK  ] DietPi-Patch | Removing obsolete DPKG exclude
[  OK  ] DietPi-Patch | Patched to DietPi v8.10
[ INFO ] DietPi-Update | APT autopurge, please wait...
Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
[  OK  ] DietPi-Update | APT autopurge
[  OK  ] DietPi-Update | Incremental patching to v8.10.2 completed
[ INFO ] DietPi-Update | Checking for new available live patches
[  OK  ] DietPi-Update | eval echo 1 > /boot/dietpi/.install_stage

 DietPi-Update
─────────────────────────────────────────────────────
 Phase: Completed

[ INFO ] DietPi-Update | Current version : v8.10.2
[ INFO ] DietPi-Update | Latest version  : v8.10.2
[  OK  ] DietPi-Survey | Setting in /boot/dietpi.txt adjusted: SURVEY_OPTED_IN=0
[  OK  ] DietPi-Survey | Purging survey data
[ INFO ] DietPi-Update | A reboot is done to finalise the kernel upgrade
Connection to 192.xxx.xxx.127 closed by remote host.
Connection to 192.xxxx.xxx.127 closed.
```
</details>

**Comment by Wilford888** [comment]
One other strange thing:  this is a Rockport64 board.  And I've checked twice to make sure I downloaded and flashed the Rockport64 image to the eMMC.  But the log file reports this as a Quartz64; it is not.

**Comment by Wilford888** [comment]
Tried the following:  I just received a QuartzA board, so I downloaded the image for that, flashed the eMMC. I was able to log in to that.  And that image is 8.10.  I think you have a problem with your current Rockport64 image on your site.

**Comment by Joulinar** [comment]
I guess issue is more with Armbian kernel / U-boot update done in parallel to initial setup / update

**Comment by canabino** [comment]
Date | Sat Oct 29 19:56:37 BST 2022
DietPi version | v8.9.2 (MichaIng/master)
Image creator | DietPi Core Team
Pre-image | Armbian
Hardware | RockPi 4b
Distro | bullseye (ID=6)
Software title | DietPi-Update

I am having the same issue but with my RockPi4.
I have recently successfully updated to Bullseye and had no problems.
Today I ran dietpi-update because it was recomending it to update to 8.10.2
After the update it wanted to restart but I couldn’t connect on SSH. I saw that the SBC wasn’t getting an IP address on my router. I then tried to view it from the HDMI connected to my TV but the TV got no signal from the HDMI port that my RockPi4 was connected to. 
I then installed a fresh image on the eMMC. It then boots up and I can access both SSH and on the HDMI. It then goes through all the steps and updates of a fresh install and asks to reboot. Then BAM, can’t get in from HDMI or SSH because no IP address gets assigned. It is connected via LAN cable.

**Comment by Wilford888** [comment]
Joulinar, what do you mean?  Is there a solution or something else I might try?  Please note, I am not having this issue with a QuartzA board.  Only with the Rockport64.

**Comment by Joulinar** [comment]
I don't have a solution yet. We would need to wait on the developer. But I see an Armbian kernel update on your logs that might have an influence. 

```
Get:2 http://armbian.tnahosting.net/apt bullseye/main arm64 linux-dtb-current-rockchip64 arm64 22.08.8 [380 kB]
Get:1 http://armbian.chi.auroradev.org/apt bullseye/main arm64 armbian-firmware all 22.08.6 [9409 kB]
Get:3 http://armbian.tnahosting.net/apt bullseye/main arm64 linux-image-current-rockchip64 arm64 22.08.8 [53.2 MB]
Get:4 http://armbian.tnahosting.net/apt bullseye/main arm64 linux-u-boot-rockpro64-current arm64 22.08.6 [436 kB]
```

**Comment by Wilford888** [comment]
Having another issue with the Quartz64 board.  I can get past the boot but when I go to install Pi-hole, I get stuck in an endless loop because I already set the static IP address.  I keep getting asked to set a static IP.  If I hit cancel, installation of Pi-hole aborts.  If I hit okay, then I circle through do I want wifi removed, IP address has changed, please reboot.  Over and over, Pi-hole never gets installed.  I've been running Pi-hole for some months.  I really wish I could just downgrade to the prior release.

**Comment by Wilford888** [comment]
I don't know if you have a QuartzA board handy.  If you do, install 8.10 and then try installing Pi-hole.  If you succeed, would really love to know what you did differently.

**Comment by Wilford888** [comment]
To replicate my issue, set a static IP prior to trying to install Pi-Hole.

**Comment by Wilford888** [comment]
The current Rockport64 image on your site is 8.9.2. Not sure if that is intentional or not.

**Comment by Joulinar** [comment]
We don't update all the images on every release. Therefore it's fine to start on a older version as it will be updated on first boot automatically.

Pls don't mix the PiHole issue in. This has nothing to do with Tue original issue reported. I guess you are already inside the PiHole install script. This is an external script managed by PiHole guy's and we just load it. Inside the install process you need to confirm you have static IP. This is how PiHole guy's designed the process now. Or are you talking about another menu?

**Comment by Wilford888** [comment]
Sorry for bringing up a different issue.  I will leave this with the original issue and check back from time to time.  Thank you.

**Comment by Joulinar** [comment]
Pls open a new issue ff you have challenges on PiHole install process.

**Comment by MichaIng** [comment]
You mean you use a "ROCKPro64", right? "Rockport64" is not an existing SBC 😉.

Is there an SD card attached to this board? Your bug report shows Quartz64 including the Quartz64 kernel, but the update log shows ROCKPro64 kernel/firmware upgraded. The only reason for such I can imagine is when bootloader/boot partition and root partition do not match, e.g. when a Quartz64 image is booted from SD card but then loading a ROCKPro64 rootfs from eMMC, or the other way round. Not sure how this is possible, though, without manually manipulating the kernel command-line parameters, since the rootfs should be always defined via UUID on our images 🤔.

**Comment by Joulinar** [comment]
@MichaIng we have a similar report on our forum. https://dietpi.com/forum/t/cant-access-dietpi-after-latest-update/14781

It seems to happen after latest update. Maybe the kernel update has an influence.

**Comment by MichaIng** [comment]
The issue with latest rockchip64 kernel has been recognised, taken down from the repo and fix is on the way: https://forum.armbian.com/topic/24196-rock-pi-4b-boot-failed-after-update/

**Comment by MichaIng** [comment]
Issue shouldn't happen anymore, kernel v5.15.75 was the faulty one, now v5.15.76 (or later) is shipped which didn't causs the issue anymore.

Still doesn't explain the Quartz64 incidence, but that is unrelated to the kernel bug. So you can now reflash and won't be hit by the bug anymore.

**Comment by Joulinar** [comment]
I guess we can close this issue, right? Because Armbian guys provided a new kernel version.
